src/api/spec/jobs/expire_assignments_job_spec.rb

```diff
@@ -0,0 +1,17 @@
+RSpec.describe ExpireAssignmentsJob do
+  include ActiveJob::TestHelper
+
+  describe '#perform' do
+    context 'when the assignment happened more than 24h ago' do
+      let!(:assignment) { travel_to(2.days.ago) { create(:assignment) } }
+
+      it { expect { ExpireAssignmentsJob.new.perform }.to change(Assignment, :count).from(1).to(0) }
+    end
+
+    context 'when the assignment did not happened more than 24h ago' do
+      let!(:assignment) { travel_to(1.hour.ago) { create(:assignment) } }
+
+      it { expect { ExpireAssignmentsJob.new.perform }.not_to change(Assignment, :count) }
+    end
+  end
+end
```

Comment:
I would remove these tests. There is not much logic involved to be checked. And test time is developer time. https://github.com/openSUSE/open-build-service/pull/17762#discussion_r2063712582